### Import Pandas library

In [1]:
import pandas as pd


### Import credits and movies dataset

In [2]:
credits = pd.read_csv("credits.csv")
movies = pd.read_csv("movies.csv")

print(" Datasets imported Successfully!!!!!!!!!!!")

 Datasets imported Successfully!!!!!!!!!!!


### Credits dataset information

In [3]:
credits.head(1)

movie_id   title                                               cast  \
0     19995  Avatar  [{"cast_id": 242, "character": "Jake Sully", "...   

                                                crew  
0  [{"credit_id": "52fe48009251416c750aca23", "de...

In [4]:
credits.shape

(4803, 4)

In [5]:
credits.info()

<class 'pandas.core.frame.DataFrame'>
RangeIndex: 4803 entries, 0 to 4802
Data columns (total 4 columns):
 #   Column    Non-Null Count  Dtype 
---  ------    --------------  ----- 
 0   movie_id  4803 non-null   int64 
 1   title     4803 non-null   object
 2   cast      4803 non-null   object
 3   crew      4803 non-null   object
dtypes: int64(1), object(3)
memory usage: 150.2+ KB


In [6]:
credits.isnull().sum()

movie_id    0
title       0
cast        0
crew        0
dtype: int64

In [7]:
credits.duplicated().sum()

0

### Movies dataset information

In [8]:
movies.head(1)

budget                                             genres  \
0  237000000  [{"id": 28, "name": "Action"}, {"id": 12, "nam...   

                      homepage     id  \
0  http://www.avatarmovie.com/  19995   

                                            keywords original_language  \
0  [{"id": 1463, "name": "culture clash"}, {"id":...                en   

  original_title                                           overview  \
0         Avatar  In the 22nd century, a paraplegic Marine is di...   

   popularity                               production_companies  \
0  150.437577  [{"name": "Ingenious Film Partners", "id": 289...   

                                production_countries release_date     revenue  \
0  [{"iso_3166_1": "US", "name": "United States o...   2009-12-10  2787965087   

   runtime                                   spoken_languages    status  \
0    162.0  [{"iso_639_1": "en", "name": "English"}, {"iso...  Released   

                       tagline   title  vote_average  vote_count  
0  Enter the World of Pandora.  Avatar           7.2       11800

In [9]:
movies.shape

(4803, 20)

In [10]:
movies.info()

<class 'pandas.core.frame.DataFrame'>
RangeIndex: 4803 entries, 0 to 4802
Data columns (total 20 columns):
 #   Column                Non-Null Count  Dtype  
---  ------                --------------  -----  
 0   budget                4803 non-null   int64  
 1   genres                4803 non-null   object 
 2   homepage              1712 non-null   object 
 3   id                    4803 non-null   int64  
 4   keywords              4803 non-null   object 
 5   original_language     4803 non-null   object 
 6   original_title        4803 non-null   object 
 7   overview              4800 non-null   object 
 8   popularity            4803 non-null   float64
 9   production_companies  4803 non-null   object 
 10  production_countries  4803 non-null   object 
 11  release_date          4802 non-null   object 
 12  revenue               4803 non-null   int64  
 13  runtime               4801 non-null   float64
 14  spoken_languages      4803 non-null   object 
 15  status               

In [11]:
movies.isnull().sum()

budget                     0
genres                     0
homepage                3091
id                         0
keywords                   0
original_language          0
original_title             0
overview                   3
popularity                 0
production_companies       0
production_countries       0
release_date               1
revenue                    0
runtime                    2
spoken_languages           0
status                     0
tagline                  844
title                      0
vote_average               0
vote_count                 0
dtype: int64

In [12]:
movies.duplicated().sum()

0

### Summary of both datasets 

### Credits :
#### shape : (4803 , 4 ) | Null values = 0 | Duplcate values = 0


### Movies:
#### shape : (4803 , 20 ) | Null values = 3935 | Duplcate values = 0

### Merge both the datasets 

In [13]:
movies_dataset = movies.merge(credits ,on="title")

In [14]:
movies_dataset.sample(1)

budget                                             genres  \
2719       0  [{"id": 35, "name": "Comedy"}, {"id": 18, "nam...   

                               homepage     id  \
2719  http://www.theluckyonesmovie.com/  16005   

                                               keywords original_language  \
2719  [{"id": 6054, "name": "friendship"}, {"id": 60...                en   

      original_title                                           overview  \
2719  The Lucky Ones  The story revolves around three soldiers — Col...   

      popularity production_companies  ... runtime  \
2719    8.747922                   []  ...   113.0   

                              spoken_languages    status  tagline  \
2719  [{"iso_639_1": "en", "name": "English"}]  Released      NaN   

               title vote_average vote_count movie_id  \
2719  The Lucky Ones          6.3         55    16005   

                                                   cast  \
2719  [{"cast_id": 4, "character": "Fred Cheaver", "...   

                                                   crew  
2719  [{"credit_id": "553e37dbc3a3687db1003063", "de...  

[1 rows x 23 columns]

In [15]:
movies_dataset.shape

(4809, 23)

In [16]:
movies_dataset.info()

<class 'pandas.core.frame.DataFrame'>
Int64Index: 4809 entries, 0 to 4808
Data columns (total 23 columns):
 #   Column                Non-Null Count  Dtype  
---  ------                --------------  -----  
 0   budget                4809 non-null   int64  
 1   genres                4809 non-null   object 
 2   homepage              1713 non-null   object 
 3   id                    4809 non-null   int64  
 4   keywords              4809 non-null   object 
 5   original_language     4809 non-null   object 
 6   original_title        4809 non-null   object 
 7   overview              4806 non-null   object 
 8   popularity            4809 non-null   float64
 9   production_companies  4809 non-null   object 
 10  production_countries  4809 non-null   object 
 11  release_date          4808 non-null   object 
 12  revenue               4809 non-null   int64  
 13  runtime               4807 non-null   float64
 14  spoken_languages      4809 non-null   object 
 15  status               

In [17]:
movies_dataset.dropna(inplace = True)

In [18]:
movies_dataset.isnull().sum()

budget                  0
genres                  0
homepage                0
id                      0
keywords                0
original_language       0
original_title          0
overview                0
popularity              0
production_companies    0
production_countries    0
release_date            0
revenue                 0
runtime                 0
spoken_languages        0
status                  0
tagline                 0
title                   0
vote_average            0
vote_count              0
movie_id                0
cast                    0
crew                    0
dtype: int64

### Remove unneccessary attributes which will not add values to our Recommender system

In [19]:
movies_dataset = movies_dataset.drop(["budget","homepage","original_language","original_title","popularity","production_companies","production_countries","release_date","revenue","runtime","spoken_languages","status","vote_count","vote_average","movie_id","tagline"] ,axis = 1)

In [20]:
movies_dataset.head(1)

genres     id  \
0  [{"id": 28, "name": "Action"}, {"id": 12, "nam...  19995   

                                            keywords  \
0  [{"id": 1463, "name": "culture clash"}, {"id":...   

                                            overview   title  \
0  In the 22nd century, a paraplegic Marine is di...  Avatar   

                                                cast  \
0  [{"cast_id": 242, "character": "Jake Sully", "...   

                                                crew  
0  [{"credit_id": "52fe48009251416c750aca23", "de...

In [21]:
movies_dataset.shape

(1494, 7)

In [22]:
movies_dataset.info()

<class 'pandas.core.frame.DataFrame'>
Int64Index: 1494 entries, 0 to 4807
Data columns (total 7 columns):
 #   Column    Non-Null Count  Dtype 
---  ------    --------------  ----- 
 0   genres    1494 non-null   object
 1   id        1494 non-null   int64 
 2   keywords  1494 non-null   object
 3   overview  1494 non-null   object
 4   title     1494 non-null   object
 5   cast      1494 non-null   object
 6   crew      1494 non-null   object
dtypes: int64(1), object(6)
memory usage: 93.4+ KB


### Now with the above attributes we will create a new attribute known as " tag " which will consist genres, cast,crew, overview and keywords

#### Task 1 : Clean Genres columns

In [23]:
movies_dataset['genres'][0]

'[{"id": 28, "name": "Action"}, {"id": 12, "name": "Adventure"}, {"id": 14, "name": "Fantasy"}, {"id": 878, "name": "Science Fiction"}]'

### From the above code cell output we cam observe a list of many dictionaries which consists id and name as a key and their detail as a value. so we can build  function which can fetch a name from every dictionary of a list of/from every movies.

In [24]:
import ast

name_list =  []
def genre_fetcher(obj):
    for i in ast.literal_eval(obj):
        name_list.append(i["name"])
    return name_list
        

In [25]:
movies_dataset["genres"] = movies_dataset["genres"].apply(genre_fetcher)
print (" Congratulations ,  Task 1 is done succesfully!!!!!!!!!!!!!! Keep it up . Lets move for task 2!!!!!")

 Congratulations ,  Task 1 is done succesfully!!!!!!!!!!!!!! Keep it up . Lets move for task 2!!!!!


In [26]:
movies_dataset["genres"]



0       [Action, Adventure, Fantasy, Science Fiction, ...
1       [Action, Adventure, Fantasy, Science Fiction, ...
2       [Action, Adventure, Fantasy, Science Fiction, ...
3       [Action, Adventure, Fantasy, Science Fiction, ...
4       [Action, Adventure, Fantasy, Science Fiction, ...
                              ...                        
4779    [Action, Adventure, Fantasy, Science Fiction, ...
4787    [Action, Adventure, Fantasy, Science Fiction, ...
4797    [Action, Adventure, Fantasy, Science Fiction, ...
4802    [Action, Adventure, Fantasy, Science Fiction, ...
4807    [Action, Adventure, Fantasy, Science Fiction, ...
Name: genres, Length: 1494, dtype: object

In [27]:
movies_dataset.head(1)

genres     id  \
0  [Action, Adventure, Fantasy, Science Fiction, ...  19995   

                                            keywords  \
0  [{"id": 1463, "name": "culture clash"}, {"id":...   

                                            overview   title  \
0  In the 22nd century, a paraplegic Marine is di...  Avatar   

                                                cast  \
0  [{"cast_id": 242, "character": "Jake Sully", "...   

                                                crew  
0  [{"credit_id": "52fe48009251416c750aca23", "de...

#### Task 2 : Clean keywords columns

In [28]:
movies_dataset['keywords'][0]

'[{"id": 1463, "name": "culture clash"}, {"id": 2964, "name": "future"}, {"id": 3386, "name": "space war"}, {"id": 3388, "name": "space colony"}, {"id": 3679, "name": "society"}, {"id": 3801, "name": "space travel"}, {"id": 9685, "name": "futuristic"}, {"id": 9840, "name": "romance"}, {"id": 9882, "name": "space"}, {"id": 9951, "name": "alien"}, {"id": 10148, "name": "tribe"}, {"id": 10158, "name": "alien planet"}, {"id": 10987, "name": "cgi"}, {"id": 11399, "name": "marine"}, {"id": 13065, "name": "soldier"}, {"id": 14643, "name": "battle"}, {"id": 14720, "name": "love affair"}, {"id": 165431, "name": "anti war"}, {"id": 193554, "name": "power relations"}, {"id": 206690, "name": "mind and soul"}, {"id": 209714, "name": "3d"}]'

### From the above code cell output we cam observe a list of many dictionaries which consists id and name as a key and their detail as a value. so we can build  a function which can fetch a name from every dictionary of a list of/from every movies

In [29]:
import ast

name_list =  []
def keyword_fetcher(obj):
    for i in ast.literal_eval(obj):
        name_list.append(i["name"])
    return name_list

In [30]:
movies_dataset['keywords'] = movies_dataset['keywords'].apply(keyword_fetcher)
print (" Congratulations ,  Task 2 is done succesfully!!!!!!!!!!!!!! Keep it up . Lets move for task 3!!!!!")

 Congratulations ,  Task 2 is done succesfully!!!!!!!!!!!!!! Keep it up . Lets move for task 3!!!!!


In [31]:
movies_dataset['keywords']

0       [culture clash, future, space war, space colon...
1       [culture clash, future, space war, space colon...
2       [culture clash, future, space war, space colon...
3       [culture clash, future, space war, space colon...
4       [culture clash, future, space war, space colon...
                              ...                        
4779    [culture clash, future, space war, space colon...
4787    [culture clash, future, space war, space colon...
4797    [culture clash, future, space war, space colon...
4802    [culture clash, future, space war, space colon...
4807    [culture clash, future, space war, space colon...
Name: keywords, Length: 1494, dtype: object

In [32]:
movies_dataset.head(1)

genres     id  \
0  [Action, Adventure, Fantasy, Science Fiction, ...  19995   

                                            keywords  \
0  [culture clash, future, space war, space colon...   

                                            overview   title  \
0  In the 22nd century, a paraplegic Marine is di...  Avatar   

                                                cast  \
0  [{"cast_id": 242, "character": "Jake Sully", "...   

                                                crew  
0  [{"credit_id": "52fe48009251416c750aca23", "de...

#### Task 3 : Clean cast columns


In [33]:
movies_dataset['cast'][0]

'[{"cast_id": 242, "character": "Jake Sully", "credit_id": "5602a8a7c3a3685532001c9a", "gender": 2, "id": 65731, "name": "Sam Worthington", "order": 0}, {"cast_id": 3, "character": "Neytiri", "credit_id": "52fe48009251416c750ac9cb", "gender": 1, "id": 8691, "name": "Zoe Saldana", "order": 1}, {"cast_id": 25, "character": "Dr. Grace Augustine", "credit_id": "52fe48009251416c750aca39", "gender": 1, "id": 10205, "name": "Sigourney Weaver", "order": 2}, {"cast_id": 4, "character": "Col. Quaritch", "credit_id": "52fe48009251416c750ac9cf", "gender": 2, "id": 32747, "name": "Stephen Lang", "order": 3}, {"cast_id": 5, "character": "Trudy Chacon", "credit_id": "52fe48009251416c750ac9d3", "gender": 1, "id": 17647, "name": "Michelle Rodriguez", "order": 4}, {"cast_id": 8, "character": "Selfridge", "credit_id": "52fe48009251416c750ac9e1", "gender": 2, "id": 1771, "name": "Giovanni Ribisi", "order": 5}, {"cast_id": 7, "character": "Norm Spellman", "credit_id": "52fe48009251416c750ac9dd", "gender": 

### From the above code cell output we cam observe a list of many dictionaries which consists id,job,name  etc as a key and their detail as a value. so we can build a function which can fetch a name of top 3 casts from every dictionary of a list of/from every movies

In [34]:
import ast


def cast_fetcher(obj):
    cast_name = []
    counter = 0
    for i in ast.literal_eval(obj):
        if counter != 3:
            cast_name.append(i["name"])
            counter +=1
        else:
            break
    return cast_name

In [35]:
movies_dataset['cast'] = movies_dataset['cast'].apply(cast_fetcher)
print (" Congratulations ,  Task 3 is done succesfully!!!!!!!!!!!!!! Keep it up . Lets move for task 4!!!!!")

 Congratulations ,  Task 3 is done succesfully!!!!!!!!!!!!!! Keep it up . Lets move for task 4!!!!!


In [36]:
movies_dataset['cast']

0        [Sam Worthington, Zoe Saldana, Sigourney Weaver]
1           [Johnny Depp, Orlando Bloom, Keira Knightley]
2            [Daniel Craig, Christoph Waltz, Léa Seydoux]
3            [Christian Bale, Michael Caine, Gary Oldman]
4          [Taylor Kitsch, Lynn Collins, Samantha Morton]
                              ...                        
4779       [Brian O'Halloran, Jeff Anderson, Jason Mewes]
4787         [Suzi Lorraine, Jared Degado, Heather Dorff]
4797    [Michael Parle, Emma Eliza Regan, Patrick O'Do...
4802        [Shane Carruth, David Sullivan, Casey Gooden]
4807            [Daniel Henney, Eliza Coupe, Bill Paxton]
Name: cast, Length: 1494, dtype: object

In [37]:
movies_dataset.head(1)

genres     id  \
0  [Action, Adventure, Fantasy, Science Fiction, ...  19995   

                                            keywords  \
0  [culture clash, future, space war, space colon...   

                                            overview   title  \
0  In the 22nd century, a paraplegic Marine is di...  Avatar   

                                               cast  \
0  [Sam Worthington, Zoe Saldana, Sigourney Weaver]   

                                                crew  
0  [{"credit_id": "52fe48009251416c750aca23", "de...

#### Task 4 : Clean crew columns

In [38]:
movies_dataset['crew'][0]

'[{"credit_id": "52fe48009251416c750aca23", "department": "Editing", "gender": 0, "id": 1721, "job": "Editor", "name": "Stephen E. Rivkin"}, {"credit_id": "539c47ecc3a36810e3001f87", "department": "Art", "gender": 2, "id": 496, "job": "Production Design", "name": "Rick Carter"}, {"credit_id": "54491c89c3a3680fb4001cf7", "department": "Sound", "gender": 0, "id": 900, "job": "Sound Designer", "name": "Christopher Boyes"}, {"credit_id": "54491cb70e0a267480001bd0", "department": "Sound", "gender": 0, "id": 900, "job": "Supervising Sound Editor", "name": "Christopher Boyes"}, {"credit_id": "539c4a4cc3a36810c9002101", "department": "Production", "gender": 1, "id": 1262, "job": "Casting", "name": "Mali Finn"}, {"credit_id": "5544ee3b925141499f0008fc", "department": "Sound", "gender": 2, "id": 1729, "job": "Original Music Composer", "name": "James Horner"}, {"credit_id": "52fe48009251416c750ac9c3", "department": "Directing", "gender": 2, "id": 2710, "job": "Director", "name": "James Cameron"},

###  From the above code cell output we cam observe a list of many dictionaries which consists id,job,name  etc as a key and their detail as a value. so we can build a function which can fetch a name of  director from every dictionary of a list of/from every movies

In [39]:
import ast


def fetch_director(text):
    dire_name = []
    for i in ast.literal_eval(text):
        if i['job'] == 'Director':
            dire_name.append(i['name'])
    return dire_name

In [40]:
 movies_dataset['crew'] = movies_dataset['crew'].apply(fetch_director)
print (" Congratulations all task  is done succesfully!!!!!!!!!!!!!! Keep it up . Lets move for another task ")

 Congratulations all task  is done succesfully!!!!!!!!!!!!!! Keep it up . Lets move for another task 


In [41]:
movies_dataset['crew']

0           [James Cameron]
1          [Gore Verbinski]
2              [Sam Mendes]
3       [Christopher Nolan]
4          [Andrew Stanton]
               ...         
4779          [Kevin Smith]
4787         [Travis Legge]
4797        [Ivan Kavanagh]
4802        [Shane Carruth]
4807          [Daniel Hsia]
Name: crew, Length: 1494, dtype: object

In [42]:
movies_dataset.head(1)

genres     id  \
0  [Action, Adventure, Fantasy, Science Fiction, ...  19995   

                                            keywords  \
0  [culture clash, future, space war, space colon...   

                                            overview   title  \
0  In the 22nd century, a paraplegic Marine is di...  Avatar   

                                               cast             crew  
0  [Sam Worthington, Zoe Saldana, Sigourney Weaver]  [James Cameron]

In [43]:
movies_dataset['overview'][0]

'In the 22nd century, a paraplegic Marine is dispatched to the moon Pandora on a unique mission, but becomes torn between following orders and protecting an alien civilization.'

In [44]:
movies_dataset.head(1)

genres     id  \
0  [Action, Adventure, Fantasy, Science Fiction, ...  19995   

                                            keywords  \
0  [culture clash, future, space war, space colon...   

                                            overview   title  \
0  In the 22nd century, a paraplegic Marine is di...  Avatar   

                                               cast             crew  
0  [Sam Worthington, Zoe Saldana, Sigourney Weaver]  [James Cameron]

### From the above code cell output we cam observe a strings of many characters ( text ) so we can build a function which can fetch a useful stemmed text from every overview of/from every movies

In [45]:
movies_dataset["overview"] = movies_dataset["overview"].apply(lambda x : x.split())

In [46]:
movies_dataset["overview"][0]

['In',
 'the',
 '22nd',
 'century,',
 'a',
 'paraplegic',
 'Marine',
 'is',
 'dispatched',
 'to',
 'the',
 'moon',
 'Pandora',
 'on',
 'a',
 'unique',
 'mission,',
 'but',
 'becomes',
 'torn',
 'between',
 'following',
 'orders',
 'and',
 'protecting',
 'an',
 'alien',
 'civilization.']

In [47]:
movies_dataset.head(1)

genres     id  \
0  [Action, Adventure, Fantasy, Science Fiction, ...  19995   

                                            keywords  \
0  [culture clash, future, space war, space colon...   

                                            overview   title  \
0  [In, the, 22nd, century,, a, paraplegic, Marin...  Avatar   

                                               cast             crew  
0  [Sam Worthington, Zoe Saldana, Sigourney Weaver]  [James Cameron]

### Eliminate the white space 

In [48]:
def collapse(L):
    L1 = []
    for i in L:
        L1.append(i.replace(" ",""))
    return L1

In [49]:
movies_dataset['cast'] = movies_dataset['cast'].apply(collapse)
movies_dataset['crew'] = movies_dataset['crew'].apply(collapse)
movies_dataset['genres'] = movies_dataset['genres'].apply(collapse)
movies_dataset['keywords'] = movies_dataset['keywords'].apply(collapse)

In [50]:
movies_dataset.head(1)

genres     id  \
0  [Action, Adventure, Fantasy, ScienceFiction, A...  19995   

                                            keywords  \
0  [cultureclash, future, spacewar, spacecolony, ...   

                                            overview   title  \
0  [In, the, 22nd, century,, a, paraplegic, Marin...  Avatar   

                                            cast            crew  
0  [SamWorthington, ZoeSaldana, SigourneyWeaver]  [JamesCameron]

## Create tag attribute

In [51]:
movies_dataset['tags'] = movies_dataset['genres'] +  movies_dataset['keywords'] +  movies_dataset['cast'] + movies_dataset['crew'] +  movies_dataset['overview']

In [52]:
movies_dataset.head(1)

genres     id  \
0  [Action, Adventure, Fantasy, ScienceFiction, A...  19995   

                                            keywords  \
0  [cultureclash, future, spacewar, spacecolony, ...   

                                            overview   title  \
0  [In, the, 22nd, century,, a, paraplegic, Marin...  Avatar   

                                            cast            crew  \
0  [SamWorthington, ZoeSaldana, SigourneyWeaver]  [JamesCameron]   

                                                tags  
0  [Action, Adventure, Fantasy, ScienceFiction, A...

In [53]:
movies_dataset = movies_dataset.drop(columns=['overview','genres','keywords','cast','crew'])
movies_dataset.head(1)

id   title                                               tags
0  19995  Avatar  [Action, Adventure, Fantasy, ScienceFiction, A...

In [54]:
movies_dataset['tags'] = movies_dataset['tags'].apply(lambda x: " ".join(x)) 
movies_dataset.head()

id                                     title  \
0   19995                                    Avatar   
1     285  Pirates of the Caribbean: At World's End   
2  206647                                   Spectre   
3   49026                     The Dark Knight Rises   
4   49529                               John Carter   

                                                tags  
0  Action Adventure Fantasy ScienceFiction Advent...  
1  Action Adventure Fantasy ScienceFiction Advent...  
2  Action Adventure Fantasy ScienceFiction Advent...  
3  Action Adventure Fantasy ScienceFiction Advent...  
4  Action Adventure Fantasy ScienceFiction Advent...

### Vectorization 

In [55]:
from sklearn.feature_extraction.text import CountVectorizer
cv = CountVectorizer(max_features=5000,stop_words='english')

In [56]:
vector = cv.fit_transform(movies_dataset['tags']).toarray() 

In [57]:
vector.shape

(1494, 5000)

In [58]:
from nltk.stem.porter import PorterStemmer
ps = PorterStemmer()

In [59]:
def stem(text):
    y = []
    for i in text.split():
        y.append(ps.stem(i))
    return " ".join(y)

In [60]:
movies_dataset['tags'].apply(stem)
movies_dataset['tags'][0]

'Action Adventure Fantasy ScienceFiction Adventure Fantasy Action Action Adventure Crime Action Crime Drama Thriller Action Adventure ScienceFiction Fantasy Action Adventure Animation Family Action Adventure ScienceFiction Adventure Fantasy Family Action Adventure Fantasy Adventure Action Thriller Crime Adventure Fantasy Action Action Adventure Western Action Adventure Fantasy ScienceFiction ScienceFiction Action Adventure Adventure Action Fantasy Action Comedy ScienceFiction Action Adventure Fantasy Action Adventure Fantasy Action Adventure Adventure Fantasy Adventure Fantasy Drama Romance Thriller Adventure Action ScienceFiction Action Adventure ScienceFiction Thriller Action Adventure Thriller Action Adventure Fantasy Action Adventure ScienceFiction Family Fantasy Adventure ScienceFiction Action Adventure ScienceFiction Action Adventure Fantasy Adventure Family Action Adventure Fantasy Adventure Action ScienceFiction Animation Family Adventure Comedy Adventure Action Thriller Scienc

In [61]:
movies_dataset.head(1)

id   title                                               tags
0  19995  Avatar  Action Adventure Fantasy ScienceFiction Advent...

In [62]:
from sklearn.metrics.pairwise import cosine_similarity

In [63]:
similarity = cosine_similarity(vector)

In [64]:
movies_dataset[movies_dataset['title'] == 'The Lego Movie'].index[0]

744

In [65]:
def recommend(movie):
    index = movies_dataset[movies_dataset['title'] == movie].index[0]
    distances = sorted(list(enumerate(similarity[index])),reverse=True,key = lambda x: x[1])
    for i in distances[1:6]:
        print(movies_dataset.iloc[i[0]].title)

In [66]:
recommend("Avatar")

Monster House
This Is It
Undiscovered
Hostel
Allegiant


### Pickle the model


In [67]:
import pickle

In [68]:
pickle.dump(movies_dataset,open('movie_list.pkl','wb'))
pickle.dump(similarity,open('similarity.pkl','wb'))